server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/NodeReplacementAllocationDecider.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.routing.allocation.decider;
+
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.routing.RoutingNode;
+import org.elasticsearch.cluster.routing.ShardRouting;
+import org.elasticsearch.cluster.routing.allocation.RoutingAllocation;
+
+public class NodeReplacementAllocationDecider extends AllocationDecider {
+
+    public static final String NAME = "node_replacement";
+
+    private static final Decision NO_REPLACEMENTS = Decision.single(Decision.Type.YES, NAME,
+        "no node replacements are currently ongoing, allocation is allowed");
+
+    @Override
+    public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, RoutingAllocation allocation) {
```

Comment:
If we want an option to move shards to all nodes during a replace, I think it would be possible to do so in `canAllocate` (here), as long as `BalancedShardsAllocator.Balancer.decideMove` runs through all nodes when deciding whether/where to force-allocate?